src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -256,7 +256,17 @@ def list_containers(client, resource_group_name, vault_name, container_type="Azu
     return _get_containers(client, container_type, status, resource_group_name, vault_name)
 
 
-def check_protection_enabled_for_vm(cmd, vm_id):
+def check_protection_enabled_for_vm(cmd, vm_id=None, vm=None, resource_group_name=None):
+    if vm_id is None:
+        if is_valid_resource_id(vm):
+            vm_id = vm
+        else:
+            vm_id = virtual_machines_cf(cmd.cli_ctx).get(resource_group_name, vm).id
+    else:
+        logger.warning(
+            """
+            Please use --vm argument instead of --vm-id. --vm-id may get deprecated in future releases.
+            """)
```

Comment:
I think you can make use of CLI's `deprecate_info` directly in `_params.py`:
`c.argument('vm_id', deprecate_info=c.deprecate(redirect='--vm', hide=True)`
Then when `--vm-id` is used, a similar warning will be shown.